pkg/iac/scanners/terraform/parser/resolvers/remote.go

```diff
@@ -87,7 +85,13 @@ func (r *remoteResolver) download(ctx context.Context, opt Options, dst string)
 		Pwd:     opt.WorkingDir,
 		Getters: getter.Getters,
 		Mode:    getter.ClientModeAny,
-		Options: opts,
+	}
+
+	terminalPrompt := os.Getenv("GIT_TERMINAL_PROMPT")
+	if err := os.Setenv("GIT_TERMINAL_PROMPT", "0"); err != nil {
```

Comment:
does it fallback to HTTP if this environment variable is not set?

Reply:
This variable disables prompt when cloning a repository: https://git-scm.com/docs/git#Documentation/git.txt-codeGITTERMINALPROMPTcode

Reply:
I was asking about the behavior that takes place when this is set.

```shell

2024-12-05T17:34:29-07:00       DEBUG   [module resolver] Caching module        key="cb2ee13f53c538b535a73ca65544a8a2"
2024-12-05T17:34:29-07:00       DEBUG   [module resolver] Downloading module    source="github.com/xxxxx/xxxxx?ref=v1"
2024-12-05T17:34:30-07:00       ERROR   [terraform evaluator] Failed to load module. Maybe try 'terraform init'?        err="failed to download: error downloading 'https://github.com/xxxxx/xxxxx.git?ref=v1': /usr/bin/git exited with 128: Cloning into '/var/folders/x7/p59nrsgs671_264kc0pv7k1m0000gn/T/.aqua/cache/cb2ee13f53c538b535a73ca65544a8a2'...\nfatal: could not read Username for 'https://github.com': terminal prompts disabled\n"
2024-12-05T17:34:30-07:00       DEBUG   [terraform evaluator] Starting post-submodules evaluation...

```

The error message is not entirely correct as in this case the prompt is disabled because of this changes introduced in the PR but we also didn't try downloading either to realize that it isn't a valid repo. Should we do that?

Reply:
Trivy successfully clones a repository if it is public. Otherwise, if the user has not provided credentials, Trivy logs a message that contains an error: `could not read Username for “https://github.com”: terminal prompts disabled`. I didn't quite understand about the fallback to HTTP, since it's the source that points to the git repository.